### PR TITLE
[BugFix] Disable Gin for shared data mode and alter gin column into non-string type

### PR DESCRIPTION
**Pull request by srlch**
Fix two problem:
1. Disable using Gin in shared data mode.
2. Disable modify column with Gin into non-string type

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -532,6 +532,15 @@ private boolean processModifyColumn(ModifyColumnClause alterClause, OlapTable ol
 
         Column oriColumn = schemaForFinding.get(modColIndex);
 
+        for (Index index : olapTable.getIndexes()) {
+            if (index.getIndexType() == IndexDef.IndexType.GIN) {
+                if (index.getColumns().contains(oriColumn.getName()) &&
+                        !modColumn.getType().isStringType()) {
+                    throw new DdlException("Can't not modify a Column with Gin into Non-String type");
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:539]:
```suggestion
                    throw new DdlException("Cannot modify a column with GIN into non-string type");
```

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java (hunk 1 above)
Comment (srlch) [reply fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java]: Fixed it

**Review by nshangyiming — DISMISSED** [review]

**Review by packy92 — DISMISSED** [review]

**Review by packy92 — DISMISSED** [review]

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 2]
```diff
@@ -532,6 +532,15 @@ private boolean processModifyColumn(ModifyColumnClause alterClause, OlapTable ol
 
         Column oriColumn = schemaForFinding.get(modColIndex);
 
+        for (Index index : olapTable.getIndexes()) {
+            if (index.getIndexType() == IndexDef.IndexType.GIN) {
+                if (index.getColumns().contains(oriColumn.getName()) &&
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:537]: is index column name case sensitive?

File: fe/fe-core/src/test/java/com/starrocks/analysis/GINIndexTest.java [hunk 3]
```diff
@@ -83,6 +85,15 @@ public void testCheckInvertedIndex() {
                 }}, KeysType.DUP_KEYS),
                 "Only support clucene implement for now");
 
+        Config.run_mode = "shared_data";
```
Comment (wyb) [inline fe/fe-core/src/test/java/com/starrocks/analysis/GINIndexTest.java:88]: better use
```
new MockUp<RunMode>() {
            @Mock
            public RunMode getCurrentRunMode() {
                return RunMode.SHARED_DATA;
            }
        };
```

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java (hunk 2 above)
Comment (srlch) [reply fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:537]: I tested this case, it seems that this code has consider the case sensitive problem.

File: fe/fe-core/src/test/java/com/starrocks/analysis/GINIndexTest.java (hunk 3 above)
Comment (srlch) [reply fe/fe-core/src/test/java/com/starrocks/analysis/GINIndexTest.java]: Fixed it

**Review by wyb — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Pull request merged**